lib/remote-connector.js

```diff
@@ -70,16 +95,29 @@ RemoteConnector.prototype.define = function(definition) {
 };
 
 RemoteConnector.prototype.resolve = function(Model) {
-  var remotes = this.remotes;
+  var auth = this.auth,
+      remotes = this.remotes,
+      headers = this.headers,
+      headerArgs = this.headerArgs;
 
   Model.sharedClass.methods().forEach(function(remoteMethod) {
     if (remoteMethod.name !== 'Change' && remoteMethod.name !== 'Checkpoint') {
-      createProxyMethod(Model, remotes, remoteMethod);
+      headers.forEach(function(header) {
+        var already = remoteMethod.accepts.filter(function(accept) {
+          return accept.http &&
+              accept.http.source === 'header' &&
+              accept.name === header.name;
+        });
+        if (!already || already.length === 0) {
+          remoteMethod.accepts.push(header);
```

Comment:
Yeah I was iffy about it as well. Ran into this not working for a relationship method. For some odd reason the method used ended up changing halfway and losing the reference to the modified accepts array.

> I guess this isn't currently possible in strong-remoting. We could add support for this case though.

Are you referring to setting the headers? 

>  setHeaders(settings.headers, ctx.req); // ctx.req will not exist

Is the request object not guaranteed to be there?